control-plane/subcommand/install-cni/command.go

```diff
@@ -0,0 +1,369 @@
+package installcni
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/containernetworking/cni/libcni"
+	"github.com/hashicorp/consul-k8s/control-plane/cni/config"
+	"github.com/hashicorp/consul-k8s/control-plane/subcommand/common"
+	"github.com/hashicorp/consul-k8s/control-plane/subcommand/flags"
+	"github.com/hashicorp/go-hclog"
+	"github.com/mitchellh/cli"
+	"github.com/mitchellh/mapstructure"
+)
+
+const (
+	defaultName            = "consul-cni"
+	defaultType            = "consul-cni"
+	defaultCNIBinDir       = "/opt/cni/bin"
+	defaultCNINetDir       = "/etc/cni/net.d"
+	defaultMultus          = false
+	defaultKubeconfig      = "ZZZZ-consul-cni-kubeconfig"
+	defaultLogLevel        = "info"
+	defaultCNIBinSourceDir = "/bin"
+)
+
+// TODO: Add description that explains the difference between CNIConfig and installConfig
+
+// installConfig are the values by the installer when running inside a container.
+type installConfig struct {
```

Comment:
installConfig is used because the installer is running inside a container and it is doing things to the host node. With the daemonset, the host system will be mounted into the container as /host/opt/cni/bin or /host/etc/cni/net.d.